src/components/StoreChoises.js

```diff
@@ -1,27 +1,126 @@
 import React from 'react';
-import {Image, View, Text, StyleSheet, TouchableOpacity} from 'react-native';
+import {
+  Image,
+  View,
+  Text,
+  StyleSheet,
+  TouchableOpacity,
+  Alert,
+} from 'react-native';
+import Toast from 'react-native-toast-message';
 import Icon from 'react-native-vector-icons/MaterialCommunityIcons';
+import sword from '../assets/images/icon-sword.png';
+import shield from '../assets/images/icon-shield.png';
+import agility from '../assets/images/icon-agility.png';
+import coin from '../assets/images/coin.png';
+import {useAuthContext} from '../contexts/AuthContext';
+import axios from 'axios';
+
+export const StoreChoises = ({item}) => {
+  const {user, setUser} = useAuthContext();
+
+  const image = () => {
+    if (item.affected_attribute === 'def') {
+      return (
+        <Image style={styles.figure} source={shield} resizeMode="contain" />
+      );
+    }
+    if (item.affected_attribute === 'atk') {
+      return (
+        <Image style={styles.figure} source={sword} resizeMode="contain" />
+      );
+    }
+    if (item.affected_attribute === 'agi') {
+      return (
+        <Image style={styles.figure} source={agility} resizeMode="contain" />
+      );
+    }
+  };
+
+  const buy = async () => {
+    try {
+      console.log(user);
+      if (item.value <= user.gold) {
+        const newValueGold = user.gold - item.value;
+        const characterUpdated = {
+          ...user,
+          gold: newValueGold,
+          equipment: [...user.equipment, item],
+          [item.affected_attribute]:
+            user[item.affected_attribute] + item.affected_amount,
+        };
+        const response = await axios.patch(
+          'https://dws-bug-hunters-api.vercel.app/api/characters/',
+          characterUpdated,
+        );
+        console.log(response.data);
+        Toast.show({
+          type: 'success',
+          text2: 'Compra feita com sucesso! (:',
+        });
+        setUser(characterUpdated);
+      } else {
+        Toast.show({
+          type: 'error',
+          text2: 'Putz, tu não tem ouro suficiente pra comprar esse item ):',
+        });
+      }
+    } catch (error) {
+      console.log(error);
+      Toast.show({
+        type: 'error',
+        text2: 'Ooops, algo deu errado ):',
+      });
+    }
+  };
+
+  const buyItem = () => {
+    Alert.alert('', 'Tem certeza que deseja comprar este item?', [
+      {text: 'Sim', onPress: buy},
+      {text: 'Não'},
+    ]);
+  };
 
-export const StoreChoises = ({image, name, label}) => {
   return (
     <View style={styles.container}>
       <View style={styles.figureValue}>
-        <Image style={styles.figure} source={image} resizeMode="contain" />
-        <Text style={styles.value}>{label}</Text>
+        {image()}
```

Comment:
Replace this line by the created component 
 `<EffectImage atb={item.affected_atribute} />`